lms/djangoapps/courseware/tests/test_inline_analytics_integration.py

```diff
@@ -0,0 +1,424 @@
+        request.user = self.instructor
+
+        response = get_analytics_answer_dist(request)
+        self.assertEquals(response.content, 'A problem has occurred retrieving the data.')
+
+    @override_settings(ANALYTICS_DATA_URL='dummy_url',
+                       ZENDESK_URL='https://zendesk.com')
+    @patch('courseware.views.process_analytics_answer_dist')
+    @patch('courseware.views.Client')
+    def test_staff_and_url(self, mock_client, mock_process_analytics):
+
+        mock_client.return_value.modules.return_value.answer_distribution.return_value = [{}]
+
+        factory = self.factory
+        request = factory.post('', self.data)
+        request.user = self.staff
+
+        mock_process_analytics.return_value = [{'dummy': 'dummy'}]
+        response = get_analytics_answer_dist(request)
+        self.assertEquals(response, [{'dummy': 'dummy'}])
+
+    @override_settings(ANALYTICS_DATA_URL='dummy_url',
+                       ZENDESK_URL='https://zendesk.com')
+    @patch('courseware.views.process_analytics_answer_dist')
+    @patch('courseware.views.Client')
+    def test_instructor_and_url(self, mock_client, mock_process_analytics):
+
+        mock_client.return_value.modules.return_value.answer_distribution.return_value = [{}]
+
+        factory = self.factory
+        request = factory.post('', self.data)
+        request.user = self.instructor
+
+        mock_process_analytics.return_value = [{'dummy': 'dummy'}]
+        response = get_analytics_answer_dist(request)
+        self.assertEquals(response, [{'dummy': 'dummy'}])
+
+    @override_settings(ANALYTICS_DATA_URL='dummy_url',
+                       ZENDESK_URL='https://zendesk.com')
+    @patch('courseware.views.Client')
+    def test_NotFoundError(self, mock_client):
+
+        mock_client.return_value.modules.return_value.answer_distribution.side_effect = NotFoundError
+
+        factory = self.factory
+        request = factory.post('', self.data)
+        request.user = self.instructor
+
+        response = get_analytics_answer_dist(request)
+        self.assertEquals(response.status_code, 404)
+        self.assertEquals(response.content, 'There are no student answers for this problem yet; please try again later.')
+
+    @override_settings(ANALYTICS_DATA_URL='dummy_url',
+                       ZENDESK_URL='https://zendesk.com')
+    @patch('courseware.views.Client')
+    def test_InvalidRequestError(self, mock_client):
+
+        mock_client.return_value.modules.return_value.answer_distribution.side_effect = InvalidRequestError
+
+        factory = self.factory
+        request = factory.post('', self.data)
+        request.user = self.instructor
+
+        response = get_analytics_answer_dist(request)
+        self.assertEquals(response.status_code, 500)
+        self.assertEquals(response.content, 'A problem has occurred retrieving the data, to report the problem click <a href="https://zendesk.com/hc/en-us/requests/new">here</a>')
```

Comment:
You use this long string a lot, perhaps it's worth defining in setup as "self.zendesk_response" or something.